monai/transforms/spatial/array.py

```diff
@@ -3228,29 +3246,36 @@ def __call__(self, array: NdarrayOrTensor):
         )
         patches = list(zip(*patch_iterator))
         patched_image = np.array(patches[0])
-        locations = np.array(patches[1])[:, 1:, 0]  # only keep the starting location
+        del patches[0]
+        locations = np.array(patches[0])[:, 1:, 0]  # only keep the starting location
+        del patches[0]
 
-        # Filter patches
-        if self.num_patches:
-            patched_image, locations = self.filter_count(patched_image, locations)
-        elif self.threshold:
+        # Apply threshold filter before filtering by count (if threshold_first is set)
+        if self.threshold_first and self.threshold is not None:
             patched_image, locations = self.filter_threshold(patched_image, locations)
 
-        # Pad the patch list to have the requested number of patches
         if self.num_patches:
-            padding = self.num_patches - len(patched_image)
-            if padding > 0:
-                patched_image = np.pad(
-                    patched_image,
-                    [[0, padding], [0, 0]] + [[0, 0]] * len(self.patch_size),
-                    constant_values=self.pad_kwargs.get("constant_values", 0),
-                )
-                locations = np.pad(locations, [[0, padding], [0, 0]], constant_values=0)
+            # Limit number of patches
+            patched_image, locations = self.filter_count(patched_image, locations)
+            # Pad the patch list to have the requested number of patches
+            if self.threshold is None:
+                padding = self.num_patches - len(patched_image)
+                if padding > 0:
+                    patched_image = np.pad(
```

Comment:
Hi @drbeh, thanks for your update, I could understand why we applied threshold filtering first. But still a little confused here, if we both set `threshold` and `num_patches`,  and after thresholding filtering, maybe we contain 50 patches, and we set `num_patches=100`, then we may need to pad to have 100 patches which may cause these patches lower than the `threshold` we set?

Reply:
Hi @KumoLiu, that's right but these lines for padding (L3262-L3264) are never reached if threshold is set:
```python
if self.threshold is None:
```

Reply:
Hi @drbeh, I know that, but there is no hint that you can't set both `threshold` and `num_patches` at the same time, or maybe we need to explain in the docstring that the return value may not be as expected when you set them at the same time.

Reply:
I updated the docstrings with mode explanation about it.